idunn/api/geocoder.py

```diff
@@ -0,0 +1,23 @@
+from typing import List, Optional
+
+from fastapi import Query
+from pydantic import BaseModel
+
+from idunn import settings
+from ..geocoder.client import geocoder_client
+from ..geocoder.models import GeocodeJson
+
+
+class ExtraParams(BaseModel):
+    shape: dict = Query(None, title="restrict search inside of a polygon")
+
+
+def get_autocomplete(
+    extra: ExtraParams = ExtraParams(),
+    query: str = Query(..., alias="q", title="query string"),
+    lon: Optional[float] = Query(None, ge=-180, le=180, title="latitude for the focus"),
+    lat: Optional[float] = Query(None, ge=-90, le=90, title="longitude for the focus"),
+    lang: str = Query(settings["DEFAULT_LANGUAGE"], title="language"),
+    limit: int = Query(10, ge=1, title="maximum number of results"),
```

Comment:
A few useful parameters are missing here: `offset`, `type[]`, `zone_type[]`, `poi_type[]`  
see https://github.com/CanalTP/mimirsbrunn/blob/master/libs/bragi/src/routes/autocomplete.rs#L59

Reply:
I feel ashamed for this one, I just expected to add them when I was done with the core structure... :(

Reply:
I added missing fields here: https://github.com/QwantResearch/idunn/pull/105/commits/63b9f079e87279dab8cf6d748dc6c0fe0c1f120f#diff-853016d15221f48bae06f82ba029d5ae

I had a hard time trying to elegantly get a documentation of the parameters while keeping them packed in a pedantic Model (which is quite convenient to use) since it requires to anotate the constructor.